crypto/src/hash.rs

```diff
@@ -82,6 +82,27 @@ impl Hash {
     }
 }
 
+#[cfg(feature = "std")]
+impl Hash {
+    /// Convert an untyped (raw) hash into a typed hash.
+    ///
+    /// # Usage
+    ///
+    /// In many situations the `Hash` is "colored" to a specific type of
+    /// entity, so it's possible to use Rust's type system to prevent
+    /// common errors.
+    ///
+    /// However, most hashes are computed from sources in which the
+    /// type is externally imposed, such as computing the hashes of
+    /// the branches of the Merkle tree, the signatures of a block
+    /// and/or computing the hash from a raw stream of bytes.  **This
+    /// function cannot check your code for type validity, so this
+    /// must be enforced by the user**.
+    pub fn typed<T>(self) -> HashOf<T> {
```

Comment:
```suggestion
    pub fn typed_unchecked<T>(self) -> HashOf<T> {
```

can we at least have suffix `_unchecked`?

Reply:
If you like.

How about `manually_typed`?

Reply:
Actually, this gives me an idea.

Reply:
I'll rename the postfix function to `untyped_unchecked`.